src/apprt/gtk/App.zig

```diff
@@ -1256,9 +1257,22 @@ pub fn run(self: *App) !void {
         self.transient_cgroup_base = path;
     } else log.debug("cgroup isolation disabled config={}", .{self.config.@"linux-cgroup"});
 
-    // Setup our D-Bus connection for listening to settings changes,
-    // and asynchronously request the initial color scheme
-    self.initDbus();
+    // Setup color scheme notifications
+    const adw_app: *adw.Application = @ptrCast(@alignCast(self.app));
+    const style_manager: *adw.StyleManager = adw_app.getStyleManager();
+    _ = gobject.Object.signals.notify.connect(
+        style_manager.as(gobject.Object),
```

Comment:
If you skip the `.as` here, you should be able to avoid the cast in the `adwNotifyDark` as well.